Fix the 1 second black screen bug

Zebouski opened the pull request:
I personally use Linux but I hear windows users have been suffering from the dxvk version of VF with the screen going black every time they alt tab in and out of the game. Someone today found this config setting resolves this issue, and I've had multiple people test this config change and confirm the problem is indeed solved.

Ge3e commented:
I can 2nd this. I have only noticed it since NVIDIA updated their driver to 546.01 prior to this I had no issues. In addition to this I have had a massive FPS drop in general V Synced to 60FPS but only hovers around 40 FPS in world. I have a RTX 4070 so its not my card.

ericschn commented:
I can confirm this is working on RTX 3080 with 536.23 driver.
Also works fine on GTX 970, but the issue wasn't present on that card in the first place.

hannesmann commented:
The setting `d3d9.enableDialogMode` disables exclusive fullscreen. You should be able to get the same effect by setting the game to windowed mode and enabling "Maximized". 

For any NVIDIA users on Windows, does changing this setting to "Prefer native" in the control panel solve the issue? 

![](https://user-images.githubusercontent.com/1131720/223780836-0ded1157-310f-4dc4-bfdb-748d4e9ef9b3.png)

Ge3e commented:
> I can confirm this is working on RTX 3080 with 536.23 driver. Also works fine on GTX 970, but the issue wasn't present on that card in the first place.

I had not issues with the 536.23 driver only latest ones eg  546.01

Ge3e commented:
> The setting `d3d9.enableDialogMode` disables exclusive fullscreen. You should be able to get the same effect by setting the game to windowed mode and enabling "Maximized".
> 
> For any NVIDIA users on Windows, does changing this setting to "Prefer native" in the control panel solve the issue?
> 
> ![](https://user-images.githubusercontent.com/1131720/223780836-0ded1157-310f-4dc4-bfdb-748d4e9ef9b3.png)

Hi @hannesmann, unfortunately no. I just tried it now.

Ge3e commented:
@Zebouski your dxvk.conf file seems to have fixed the issue. Thanks.

# DXVK configuration for World of Warcraft 1.12

d3d9.enableDialogMode = True

# Uncomment to set framerate limit
# d3d9.maxFrameRate = 1000

# Disabled because it causes out of memory errors
dxvk.enableGraphicsPipelineLibrary = False
dxvk.numCompilerThreads = 2
dxvk.numCompilerThreads = 2

hannesmann commented:
> Hi @hannesmann, unfortunately no. I just tried it now.

Hmm, well I'm not sure what else could have changed in a driver update (you could try the other option too). I don't want to merge this because it would disable exclusive fullscreen for AMD/Intel users but I updated the guidance on the releases page to mention the fix.

> Still have low FPS though on an ASUS 4070 RTX - i9 9900k. With everything maxed out I get 40-55 FPS. Should be couple of hundred at least.

Could you try removing DXVK (d3d9.dll) and see if that helps?

Zebouski commented:
> Hmm, well I'm not sure what else could have changed in a driver update (you could try the other option too). I don't want to merge this because it would disable exclusive fullscreen for AMD/Intel users but I updated the guidance on the releases page to mention the fix.

Thanks, I didn't know this was an issue only affecting nvidia+windows, and that going borderless full windowed does the same thing as this config value. I've updated the turtle wow wiki to direct people to your instructions. I think you can close this PR now and move people to a github issue for further troubleshooting.

But yeah from what I've heard from others, I thought the general recommendation these days for 1.12 was to not use exclusive fullscreen at all? Or is that another thing that is only an issue for certain brand hardware?

hannesmann commented:
> I thought the general recommendation these days for 1.12 was to not use exclusive fullscreen at all?

The last time I tried it exclusive fullscreen was the only way to get FreeSync working on AMD. Since this bug only seems to have appeared with 546.01 I'll assume it's an issue specific to NVIDIA for now. 

Closing in favor of #36
